client/me/purchases/pre-cancelation-dialog/index.tsx

```diff
@@ -0,0 +1,180 @@
+	site: SiteExcerptData;
+	hasDomain: boolean;
+	isRefundable: boolean;
+	isAutoRenewing: boolean;
+	primaryDomain: string;
+	wpcomURL: string;
+}
+
+export const PreCancelationDialog = ( {
+	closeDialog,
+	removePlan,
+	isDialogVisible,
+	site,
+	hasDomain,
+	isRefundable,
+	isAutoRenewing,
+	primaryDomain,
+	wpcomURL,
+}: PreCancelationDialogProps ) => {
+	const translate = useTranslate();
+
+	/**
+	 * Istantiate site's plan variables.
+	 */
+	const productSlug = site.plan?.product_slug;
+	const planLabel = site.plan?.product_name_short;
+	const subTitle = translate( 'Subtitle' );
+
+	/**
+	 * Click events, buttons tracking and action.
+	 */
+	const clickCancelPlan = () => {
+		removePlan();
+	};
+
+	const clickCloseDialog = () => {
+		closeDialog();
+	};
+
+	/**
+	 * Domain redirect copy
+	 */
+	const domainFeature =
+		hasDomain && primaryDomain && wpcomURL ? (
+			<p>Warning: domain traffic will be redirected</p>
+		) : null;
+
+	/**
+	 * Dialog buttons
+	 */
+	const buttons = [
+		{
+			action: 'cancel',
+			label: translate( 'Cancel my plan' ),
+			onClick: clickCancelPlan,
+		},
+		{
+			action: 'close',
+			label: translate( 'Keep my plan' ),
+			onClick: clickCloseDialog,
+			isPrimary: true,
+		},
+	];
+
+	/**
+	 * Return the list of features that the user will lose by canceling their plan.
+	 *
+	 * @returns Fragment
+	 */
+	const FeaturesList = () => {
+		if ( typeof productSlug === 'string' ) {
+			const planFeatures = getPlanFeatures( productSlug );
+
+			return (
+				<>
+					<p>{ subTitle }</p>
+					<ul
+						className={
+							'remove-plan-dialog__list-plan-features' +
+							( domainFeature ? ' --with-domain-feature' : '' )
+						}
+					>
+						{ domainFeature && (
+							<li key="redirect-domain">
+								<Gridicon
+									className="remove-plan-dialog__item-cross-small"
+									size={ 24 }
+									icon="cross-small"
+								/>
+								{ domainFeature }
+							</li>
+						) }
+						<li key="debug-refundable">
+							<Gridicon
+								className="remove-plan-dialog__item-cross-small"
+								size={ 24 }
+								icon="cross-small"
+							/>
+							{ isRefundable ? 'Refundable: yes' : 'Refundable: no' }
+						</li>
+						<li key="debug-autorenew">
+							<Gridicon
+								className="remove-plan-dialog__item-cross-small"
+								size={ 24 }
+								icon="cross-small"
+							/>
+							{ isAutoRenewing ? 'Auto-renew: yes' : 'Auto-renew: no' }
+						</li>
+						{ planFeatures.map( ( feature ) => {
+							return (
+								<li key={ feature }>
+									<Gridicon
+										className="remove-plan-dialog__item-cross-small"
+										size={ 24 }
+										icon="cross-small"
+									/>
+									{ feature }
+								</li>
+							);
+						} ) }
+					</ul>
+				</>
+			);
+		}
+
+		return null;
+	};
+
+	/**
+	 * Dialog classname
+	 */
+	const classDomainFeature = domainFeature ? ' --with-domain-feature' : '';
+	const dialogClassName = 'remove-plan-dialog' + classDomainFeature;
```

Comment:
FYI we have [the classnames package](https://www.npmjs.com/package/classnames) that can make this kind of conditional classname computation a lot easier, e.g. https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/intro/index.tsx#L19 and https://github.com/Automattic/wp-calypso/blob/trunk/client/components/marketing-message/index.tsx#L90 . It's up to you, of course :)

Reply:
Thanks for letting me kno, the dialog now makes use of classnames package.